src/Essentials/src/AppInfo/AppInfo.android.cs

```diff
@@ -44,14 +42,21 @@ public void ShowSettingsUI()
 			context.StartActivity(settingsIntent);
 		}
 
-		static AppTheme GetRequestedTheme() => (Application.Context.Resources.Configuration.UiMode & UiMode.NightMask) switch
+		static AppTheme GetRequestedTheme()
 		{
-			UiMode.NightYes => AppTheme.Dark,
-			UiMode.NightNo => AppTheme.Light,
-			_ => AppTheme.Unspecified
-		};
+			var config = Application.Context.Resources?.Configuration;
```

Comment:
Can we still cache the result of `Application.Context.Resources?.Configuration`? It's also in this file twice.

We could also consider `Application.Context.Resources` if we can't do the `Configuration`.

Reply:
Just FYI the source of `Application.Context` is:

https://github.com/xamarin/xamarin-android/blob/0ea65a0cf0d48833714f6ca824edc8ba43427859/src/Mono.Android/Android.App/Application.cs#L11-L26

Reply:
I think it changes tbh:

> Return the **current configuration** that is in effect for this resource object. The returned object should be treated as read-only.

https://developer.android.com/reference/android/content/res/Resources#getConfiguration()

I also see https://developer.android.com/reference/android/content/res/Resources#updateConfiguration(android.content.res.Configuration,%20android.util.DisplayMetrics)

Reply:
You have to go to the source code:

https://github.com/aosp-mirror/platform_frameworks_base/blob/2fbb619b537442662f14c7f36fb794f12250be2e/core/java/android/app/ResourcesManager.java#L317-L321

It looks like same instance of `mResConfiguration` always, but we should just test a configuration change to be sure it is ok.

Reply:
I will do another PR to cache the context and resources in essentials as it is used all over.

The trick is, does it change with instrumentation and/or different things - like services? I assume not as there is only ever 1 Context and that ever only has 1 Resources instance? Do we have docs/proof on this before I cache things?

Reply:
The Xamarin.Android code above caches the one `Application.Context`. It does appear that `Resources` is always the same instance reading Android's source code.

Google probably isn't good enough to document this.

Reply:
OK, thanks. I am working on a PR to cache the Resources, PackageManager and  other things that are used all over Essentials. This might just generally improve perf for several APIs as we basically always to the Context.Resources or Context.PackageManager in every method.